python_modules/dagster/dagster/_core/execution/plan/global_concurrency_context.py

```diff
@@ -0,0 +1,102 @@
+import time
+from collections import defaultdict
+from types import TracebackType
+from typing import (
+    List,
+    Set,
+    Type,
+)
+
+from typing_extensions import Self
+
+from dagster._core.instance import DagsterInstance
+
+CONCURRENCY_CLAIM_BLOCKED_INTERVAL = 1
+
+
+class GlobalConcurrencyContext:
```

Comment:
this doc block would be a good place to overview the setup for how this all works